### PR TITLE
Add volume for Sudo Finance

### PR DESCRIPTION
**Pull request by sudofina**
Add Sudo Finance Volume

### CODE REVIEW

**Comment by llamatester** [comment]
The sudo-fina adapter exports: 
        
 
 _______________________________________
    Dexs for 6/2/2024
    _______________________________________
    
    Failed to get value on sui: AxiosError: Request failed with status code 404
    SUI 👇
    Timestamp attempted: 5/2/2024
    AxiosError: Request failed with status code 404
        at settle (/home/runner/work/dimension-adapters/dimension-adapters/node_modules/axios/lib/core/settle.js:19:12)
        at IncomingMessage.handleStreamEnd (/home/runner/work/dimension-adapters/dimension-adapters/node_modules/axios/lib/adapters/http.js:589:11)
        at IncomingMessage.emit (node:events:529:35)
        at IncomingMessage.emit (node:domain:489:12)
        at endReadableNT (node:internal/streams/readable:1400:12)
        at processTicksAndRejections (node:internal/process/task_queues:82:21)
        at Axios.request (/home/runner/work/dimension-adapters/dimension-adapters/node_modules/axios/lib/core/Axios.js:45:41)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
        at async retries (/home/runner/work/dimension-adapters/dimension-adapters/utils/fetchURL.ts:5:35)
        at async /home/runner/work/dimension-adapters/dimension-adapters/dexs/sudo-fina/index.ts:25:30
        at async /home/runner/work/dimension-adapters/dimension-adapters/adapters/utils/runAdapter.ts:32:52
        at async Promise.allSettled (index 0)
        at async /home/runner/work/dimension-adapters/dimension-adapters/cli/testAdapter.ts:52:23 {
      code: 'ERR_BAD_REQUEST',
      config: {
        transitional: {
          silentJSONParsing: true,
          forcedJSONParsing: true,
          clarifyTimeoutError: false
        },
        adapter: [ 'xhr', 'http' ],
        transformRequest: [ [Function: transformRequest] ],
        transformResponse: [ [Function: transformResponse] ],
        timeout: 0,
        xsrfCookieName: 'XSRF-TOKEN',
        xsrfHeaderName: 'X-XSRF-TOKEN',
        maxContentLength: -1,
        maxBodyLength: -1,
        env: { FormData: [Function], Blob: [class Blob] },
        validateStatus: [Function: validateStatus],
        headers: Object [AxiosHeaders] {
          Accept: 'application/json, text/plain, */*',
          'Content-Type': undefined,
          'User-Agent': 'axios/1.6.7',
          'Accept-Encoding': 'gzip, compress, deflate, br'
        },
        method: 'get',
        url: 'https://e5370tyxcb.execute-api.ap-northeast-1.amazonaws.com/volume',
        data: undefined
      },
      request: <ref *1> ClientRequest {
        _events: [Object: null prototype] {
          abort: [Function (anonymous)],
          aborted: [Function (anonymous)],
          connect: [Function (anonymous)],
          error: [Function (anonymous)],
          socket: [Function (anonymous)],
          timeout: [Function (anonymous)],
          finish: [Function: requestOnFinish]
        },
        _eventsCount: 7,
        _maxListeners: undefined,
        outputData: [],
        outputSize: 0,
        writable: true,
        destroyed: false,
        _last: true,
        chunkedEncoding: false,
        shouldKeepAlive: false,
        maxRequestsOnConnectionReached: false,
        _defaultKeepAlive: true,
        useChunkedEncodingByDefault: false,
        sendDate: false,
        _removedConnection: false,
        _removedContLen: false,
        _removedTE: false,
        strictContentLength: false,
        _contentLength: 0,
        _hasBody: true,
        _trailer: '',
        finished: true,
        _headerSent: true,
        _closed: false,
        socket: TLSSocket {
          _tlsOptions: [Object],
          _secureEstablished: true,
          _securePending: false,
          _newSessionPending: false,
          _controlReleased: true,
          secureConnecting: false,
          _SNICallback: null,
          servername: 'e5370tyxcb.execute-api.ap-northeast-1.amazonaws.com',
          alpnProtocol: false,
          authorized: true,
          authorizationError: null,
          encrypted: true,
          _events: [Object: null prototype],
          _eventsCount: 10,
          connecting: false,
          _hadError: false,
          _parent: null,
          _host: 'e5370tyxcb.execute-api.ap-northeast-1.amazonaws.com',
          _closeAfterHandlingError: false,
          _readableState: [ReadableState],
          _maxListeners: undefined,
          _writableState: [WritableState],
          allowHalfOpen: false,
          _sockname: null,
          _pendingData: null,
          _pendingEncoding: '',
          server: undefined,
          _server: null,
          ssl: [TLSWrap],
          _requestCert: true,
          _rejectUnauthorized: true,
          parser: null,
          _httpMessage: [Circular *1],
          [Symbol(alpncallback)]: null,
          [Symbol(res)]: [TLSWrap],
          [Symbol(verified)]: true,
          [Symbol(pendingSession)]: null,
          [Symbol(async_id_symbol)]: 88,
          [Symbol(kHandle)]: [TLSWrap],
          [Symbol(lastWriteQueueSize)]: 0,
          [Symbol(timeout)]: null,
          [Symbol(kBuffer)]: null,
          [Symbol(kBufferCb)]: null,
          [Symbol(kBufferGen)]: null,
          [Symbol(kCapture)]: false,
          [Symbol(kSetNoDelay)]: false,
          [Symbol(kSetKeepAlive)]: true,
          [Symbol(kSetKeepAliveInitialDelay)]: 60,
          [Symbol(kBytesRead)]: 0,
          [Symbol(kBytesWritten)]: 0,
          [Symbol(connect-options)]: [Object]
        },
        _header: 'GET /volume HTTP/1.1\r\n' +
          'Accept: application/json, text/plain, */*\r\n' +
          'User-Agent: axios/1.6.7\r\n' +
          'Accept-Encoding: gzip, compress, deflate, br\r\n' +
          'Host: e5370tyxcb.execute-api.ap-northeast-1.amazonaws.com\r\n' +
          'Connection: close\r\n' +
          '\r\n',
        _keepAliveTimeout: 0,
        _onPendingData: [Function: nop],
        agent: Agent {
          _events: [Object: null prototype],
          _eventsCount: 2,
          _maxListeners: undefined,
          defaultPort: 443,
          protocol: 'https:',
          options: [Object: null prototype],
          requests: [Object: null prototype] {},
          sockets: [Object: null prototype],
          freeSockets: [Object: null prototype] {},
          keepAliveMsecs: 1000,
          keepAlive: false,
          maxSockets: Infinity,
          maxFreeSockets: 256,
          scheduling: 'lifo',
          maxTotalSockets: Infinity,
          totalSocketCount: 1,
          maxCachedSessions: 100,
          _sessionCache: [Object],
          [Symbol(kCapture)]: false
        },
        socketPath: undefined,
        method: 'GET',
        maxHeaderSize: undefined,
        insecureHTTPParser: undefined,
        joinDuplicateHeaders: undefined,
        path: '/volume',
        _ended: true,
        res: IncomingMessage {
          _readableState: [ReadableState],
          _events: [Object: null prototype],
          _eventsCount: 4,
          _maxListeners: undefined,
          socket: [TLSSocket],
          httpVersionMajor: 1,
          httpVersionMinor: 1,
          httpVersion: '1.1',
          complete: true,
          rawHeaders: [Array],
          rawTrailers: [],
          joinDuplicateHeaders: undefined,
          aborted: false,
          upgrade: false,
          url: '',
          method: null,
          statusCode: 404,
          statusMessage: 'Not Found',
          client: [TLSSocket],
          _consuming: false,
          _dumped: false,
          req: [Circular *1],
          responseUrl: 'https://e5370tyxcb.execute-api.ap-northeast-1.amazonaws.com/volume',
          redirects: [],
          [Symbol(kCapture)]: false,
          [Symbol(kHeaders)]: [Object],
          [Symbol(kHeadersCount)]: 16,
          [Symbol(kTrailers)]: null,
          [Symbol(kTrailersCount)]: 0
        },
        aborted: false,
        timeoutCb: null,
        upgradeOrConnect: false,
        parser: null,
        maxHeadersCount: null,
        reusedSocket: false,
        host: 'e5370tyxcb.execute-api.ap-northeast-1.amazonaws.com',
        protocol: 'https:',
        _redirectable: Writable {
          _writableState: [WritableState],
          _events: [Object: null prototype],
          _eventsCount: 3,
          _maxListeners: undefined,
          _options: [Object],
          _ended: true,
          _ending: true,
          _redirectCount: 0,
          _redirects: [],
          _requestBodyLength: 0,
          _requestBodyBuffers: [],
          _onNativeResponse: [Function (anonymous)],
          _currentRequest: [Circular *1],
          _currentUrl: 'https://e5370tyxcb.execute-api.ap-northeast-1.amazonaws.com/volume',
          [Symbol(kCapture)]: false
        },
        [Symbol(kCapture)]: false,
        [Symbol(kBytesWritten)]: 0,
        [Symbol(kNeedDrain)]: false,
        [Symbol(corked)]: 0,
        [Symbol(kOutHeaders)]: [Object: null prototype] {
          accept: [Array],
          'user-agent': [Array],
          'accept-encoding': [Array],
          host: [Array]
        },
        [Symbol(errored)]: null,
        [Symbol(kHighWaterMark)]: 16384,
        [Symbol(kRejectNonStandardBodyWrites)]: false,
        [Symbol(kUniqueHeaders)]: null
      },
      response: {
        status: 404,
        statusText: 'Not Found',
        headers: Object [AxiosHeaders] {
          date: 'Wed, 07 Feb 2024 05:37:53 GMT',
          'content-type': 'text/html; charset=utf-8',
          'content-length': '145',
          connection: 'close',
          'x-powered-by': 'Express',
          'content-security-policy': "default-src 'none'",
          'x-content-type-options': 'nosniff',
          'apigw-requestid': 'SwCPyhOCtjMEMhA='
        },
        config: {
          transitional: [Object],
          adapter: [Array],
          transformRequest: [Array],
          transformResponse: [Array],
          timeout: 0,
          xsrfCookieName: 'XSRF-TOKEN',
          xsrfHeaderName: 'X-XSRF-TOKEN',
          maxContentLength: -1,
          maxBodyLength: -1,
          env: [Object],
          validateStatus: [Function: validateStatus],
          headers: [Object [AxiosHeaders]],
          method: 'get',
          url: 'https://e5370tyxcb.execute-api.ap-northeast-1.amazonaws.com/volume',
          data: undefined
        },
        request: <ref *1> ClientRequest {
          _events: [Object: null prototype],
          _eventsCount: 7,
          _maxListeners: undefined,
          outputData: [],
          outputSize: 0,
          writable: true,
          destroyed: false,
          _last: true,
          chunkedEncoding: false,
          shouldKeepAlive: false,
          maxRequestsOnConnectionReached: false,
          _defaultKeepAlive: true,
          useChunkedEncodingByDefault: false,
          sendDate: false,
          _removedConnection: false,
          _removedContLen: false,
          _removedTE: false,
          strictContentLength: false,
          _contentLength: 0,
          _hasBody: true,
          _trailer: '',
          finished: true,
          _headerSent: true,
          _closed: false,
          socket: [TLSSocket],
          _header: 'GET /volume HTTP/1.1\r\n' +
            'Accept: application/json, text/plain, */*\r\n' +
            'User-Agent: axios/1.6.7\r\n' +
            'Accept-Encoding: gzip, compress, deflate, br\r\n' +
            'Host: e5370tyxcb.execute-api.ap-northeast-1.amazonaws.com\r\n' +
            'Connection: close\r\n' +
            '\r\n',
          _keepAliveTimeout: 0,
          _onPendingData: [Function: nop],
          agent: [Agent],
          socketPath: undefined,
          method: 'GET',
          maxHeaderSize: undefined,
          insecureHTTPParser: undefined,
          joinDuplicateHeaders: undefined,
          path: '/volume',
          _ended: true,
          res: [IncomingMessage],
          aborted: false,
          timeoutCb: null,
          upgradeOrConnect: false,
          parser: null,
          maxHeadersCount: null,
          reusedSocket: false,
          host: 'e5370tyxcb.execute-api.ap-northeast-1.amazonaws.com',
          protocol: 'https:',
          _redirectable: [Writable],
          [Symbol(kCapture)]: false,
          [Symbol(kBytesWritten)]: 0,
          [Symbol(kNeedDrain)]: false,
          [Symbol(corked)]: 0,
          [Symbol(kOutHeaders)]: [Object: null prototype],
          [Symbol(errored)]: null,
          [Symbol(kHighWaterMark)]: 16384,
          [Symbol(kRejectNonStandardBodyWrites)]: false,
          [Symbol(kUniqueHeaders)]: null
        },
        data: '<!DOCTYPE html>\n' +
          '<html lang="en">\n' +
          '<head>\n' +
          '<meta charset="utf-8">\n' +
          '<title>Error</title>\n' +
          '</head>\n' +
          '<body>\n' +
          '<pre>Cannot GET /volume</pre>\n' +
          '</body>\n' +
          '</html>\n'
      }
    }
    
    
    
    
    Done in 16.05s.

**Comment by sudofina** [comment]
Retry

File: dexs/sudo-fina/index.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  weekVolume: number,
+  monthVolume: number,
+}
+
+const fetch = (chain: Chain) => {
+  return async (timestamp: number) => {
+    const dayTimestamp = getUniqStartOfTodayTimestamp(new Date(timestamp * 1000))
+    const volume: IVolume = (await fetchURL(url[chain]))?.data;
+    return {
+      totalVolume: `${volume?.totalVolume || undefined}`,
```
Comment (sudofina) [inline dexs/sudo-fina/index.ts]: volume?.totalVolume is already undefined if it doesn't exist

File: dexs/sudo-fina/index.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    };
+  };
+}
+
+const adapter: SimpleAdapter = {
+  adapter: {
+    [CHAIN.SUI]: {
+      fetch: fetch(CHAIN.SUI),
+      runAtCurrTime: true,
+      start: async () => 1682121600,
```
Comment (sudofina) [inline dexs/sudo-fina/index.ts]: did we make sure the start time is the first day we launched? 1/12 or 1/13 i think

**Comment by dtmkeng** [comment]
error is seem block by cf

**Comment by sudofina** [comment]
we don't have cloudflare and the api server was temporarily down, will resubmit soon

**Comment by llamatester** [comment]
The sudofinance adapter exports: 
        
 
 _______________________________________
    Dexs for 6/2/2024
    _______________________________________
    
    Wrong value undefined with key weekVolume
    Wrong value undefined with key monthVolume
    SUI 👇
    Backfill start time: 5/1/2024
    NO METHODOLOGY SPECIFIED
    Total volume: 2.06 M
    Daily volume: 81.23 k
    Timestamp: 1707177600
    
    
    
    
    Done in 2.83s.

**Comment by llamatester** [comment]
The sudofinance adapter exports: 
        
 
 _______________________________________
    Dexs for 6/2/2024
    _______________________________________
    
    Wrong value undefined with key weekVolume
    Wrong value undefined with key monthVolume
    SUI 👇
    Backfill start time: 5/1/2024
    NO METHODOLOGY SPECIFIED
    Total volume: 2.06 M
    Daily volume: 81.23 k
    Timestamp: 1707177600
    
    
    
    
    Done in 2.87s.

**Comment by llamatester** [comment]
The sudofinance adapter exports: 
        
 
 _______________________________________
    Dexs for 6/2/2024
    _______________________________________
    
    SUI 👇
    Backfill start time: 5/1/2024
    NO METHODOLOGY SPECIFIED
    Total volume: 2.06 M
    Daily volume: 81.23 k
    Timestamp: 1707177600
    
    
    
    
    Done in 2.96s.

**Comment by llamatester** [comment]
The sudofinance adapter exports: 
        
 
 _______________________________________
    Dexs for 6/2/2024
    _______________________________________
    
    SUI 👇
    Backfill start time: 5/1/2024
    NO METHODOLOGY SPECIFIED
    Total volume: 2.06 M
    Daily volume: 81.23 k
    Timestamp: 1707177600
    
    
    
    
    Done in 2.88s.

**Comment by sudofina** [comment]
looks like our api won't work with backfill just yet, will update

**Comment by llamatester** [comment]
The sudofinance adapter exports: 
        
 
 _______________________________________
    Dexs for 6/2/2024
    _______________________________________
    
    SUI 👇
    Backfill start time: 5/1/2024
    NO METHODOLOGY SPECIFIED
    Total volume: 1.98 M
    Daily volume: 51.63 k
    Timestamp: 1707177600
    
    
    
    
    Done in 2.89s.

**Comment by sudofina** [comment]
should be good now @dtmkeng

**Comment by llamatester** [comment]
The sudofinance adapter exports: 
        
 
 _______________________________________
    Dexs for 6/2/2024
    _______________________________________
    
    SUI 👇
    Backfill start time: 5/1/2024
    NO METHODOLOGY SPECIFIED
    Total volume: 1.98 M
    Daily volume: 51.63 k
    Timestamp: 1707177600
    
    
    
    
    Done in 2.98s.

**Comment by sudofina** [comment]
> > should be good now @dtmkeng
> 
> is it value of derivatives volume ?

yes

**Comment by sudofina** [comment]
> > should be good now @dtmkeng
> 
> is it value of derivatives volume ?

do you want me to put under derivative category?

**Comment by dtmkeng** [comment]
> > > should be good now @dtmkeng
> > 
> > 
> > is it value of derivatives volume ?
> 
> do you want me to put under derivative category?

yes

**Comment by dtmkeng** [comment]
@sudofina  could you check adapter it return error

**Comment by dtmkeng** [comment]
fixed

**Comment by sudofina** [comment]
I can make a follow up change to put the volume under derivative.

**Pull request merged**